src/analysis/normed_space/star/continuous_functional_calculus.lean

```diff
@@ -0,0 +1,235 @@
+/-
+Copyright (c) 2022 Jireh Loreaux. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jireh Loreaux
+-/
+
+import analysis.normed_space.star.gelfand_duality
+import topology.algebra.star_subalgebra
+
+/-! # Continuous functional calculus
+
+In this file we construct the `continuous_functional_calculus` for normal elements of a (unital)
+C⋆-algebra over `ℂ`. This is a star algebra equivalence
+`C(spectrum ℂ a, ℂ) ≃⋆ₐ[ℂ] elemental_star_algebra ℂ a` which sends the (restriction of) the
+identity map `continuous_map.id ℂ` to the (unique) preimage of `a` under the coercion of
+`elemental_star_algebra ℂ a` to `A`.
+
+Being a star algebra equivalence between C⋆-algebras, this map is continuous (even an isometry),
+and by the Stone-Weierstrass theorem it is the unique star algebra equivalence which extends the
+polynomial functional calculus (i.e., `polynomial.aeval`).
+
+This file also includes a proof of the **spectral permanence** theorem for (unital) C⋆-algebras
+(see `star_subalgebra.spectrum_eq`)
+
+## Main definitions
+
+* `continuous_functional_calculus : C(spectrum ℂ a, ℂ) ≃⋆ₐ[ℂ] elemental_star_algebra ℂ a`
+* `elemental_star_algebra.character_space_homeo :
+  `character_space ℂ (elemental_star_algebra ℂ a) ≃ₜ spectrum ℂ a`
+
+## Main statements
+
+* `star_subalgebra.coe_is_unit`: for `x : S` in a C⋆-subalgebra `S` of `A`, then `↑x : A` is a unit
+  if and only if `x` is a unit.
+* `star_subalgebra.spectrum_eq`: **spectral_permanence.** for `x : S`, where `S` is a C⋆-subalgebra
+  of `A`, `spectrum ℂ x = spectrum ℂ (x : A)`.
+
+## Notes
+
+The result we have established here is the strongest possible, but it is likely not the version
+which will be most useful in practice. Future work will include developing an appropriate API for
+the continuous functional calculus (including one for real-valued functions with real argument that
+applies to self-adjoint elements of the algebra). -/
+
+open_locale pointwise ennreal nnreal complex_order
+
+open weak_dual weak_dual.character_space elemental_star_algebra
+
+variables {A : Type*} [normed_ring A] [normed_algebra ℂ A]
+variables [star_ring A] [cstar_ring A] [star_module ℂ A]
+
+instance {R A : Type*} [comm_ring R] [star_ring R] [normed_ring A] [algebra R A] [star_ring A]
+  [has_continuous_star A] [star_module R A] (a : A) [is_star_normal a] :
+  normed_comm_ring (elemental_star_algebra R a) :=
+{ mul_comm := mul_comm, .. subring_class.to_normed_ring (elemental_star_algebra R a) }
```

Comment:
Is there no `subring_class.to_normed_comm_ring`?

Reply:
There is, but the algebra `A` need but be commutative, so that inheritance wouldn't work. Nevertheless, because `a` is normal, this subalgebra is commutative.